pkg/chaosdaemon/tasks/task_config_manager.go

```diff
@@ -105,25 +107,36 @@ func (m TaskConfigManager) DeleteTaskConfig(id UID) error {
 	return nil
 }
 
-func (m TaskConfigManager) GetConfigWithUID(id UID) (interface{}, error) {
+func (m TaskConfigManager) GetConfigWithUID(id UID) (TaskConfig, error) {
 	t, ok := m.TaskConfigMap[id]
 	if !ok {
 		return TaskConfig{}, ErrUIDNotFound
 	}
-	return t.data, nil
+	return t, nil
 }
 
 func (m TaskConfigManager) GetUIDsWithPID(id PID) []UID {
 	uIds := make([]UID, 0)
 	for uid, task := range m.TaskConfigMap {
-		if task.main == id {
+		if task.Main == id {
 			uIds = append(uIds, uid)
 		}
 	}
 	return uIds
 }
 
-// SumTaskConfig will sum the TaskConfig with a same TaskConfig.main.
+func (m TaskConfigManager) CheckTask(uid UID, pid PID) error {
+	t, ok := m.TaskConfigMap[uid]
+	if !ok {
+		return ErrUIDNotFound
+	}
+	if t.Main != pid {
+		return ErrDiffPID
+	}
+	return nil
+}
+
+// SumTaskConfig will sum the TaskConfig with a same TaskConfig.Main.
 // If developers want to use it with type T, they must implement Addable for *T.
 // IMPORTANT: Just here , we do not assume A.Add(B) == B.Add(A).
 // What SumTaskConfig do : A := new(TaskConfig), A.Add(B).Add(C).Add(D)... , A marked as uid.
```

Comment:
I thins `MergeTaskConfig` is a better name.

Reply:
I agree with you.

Reply:
Fixed.